crud/select/executor.lua

```diff
@@ -31,52 +30,51 @@ local function scroll_to_after_tuple(gen, param, state, space, scanner)
             return nil
         end
 
-        if scroll_comparator(tuple, scanner.after_tuple) then
+        if scroll_comparator(tuple, after_tuple) then
             return tuple
         end
     end
 end
 
-function executor.execute(plan)
-    local scanner = plan.scanner
+function executor.execute(space, index, scan_value, iter, filter_func, opts)
+    checks('table', 'table', '?table', 'number', 'function', {
+        after_tuple = '?cdata|table',
+        batch_size = '?number',
+    })
 
-    if scanner.limit == 0 then
+    opts = opts or {}
+
+    if opts.batch_size == 0 then
         return {}
     end
 
-    local filter = select_filters.gen_code(plan.filter_conditions)
-    local filer_func = select_filters.compile(filter)
-
     local tuples = {}
     local tuples_count = 0
 
-    local space = box.space[scanner.space_name]
-    local index = space.index[scanner.index_id]
-
-    local scan_value = scanner.value
-    if scanner.after_tuple ~= nil then
-        if scan_value == nil then
-            scan_value = scanner.after_tuple
+    local value = scan_value
+    if opts.after_tuple ~= nil then
+        if value == nil then
+            value = opts.after_tuple
         else
-            local cmp_operator = select_comparators.get_cmp_operator(scanner.iter)
+            local cmp_operator = select_comparators.get_cmp_operator(iter)
             local scan_comparator, err = select_comparators.gen_tuples_comparator(cmp_operator, index.parts)
             if err ~= nil then
                 log.warn("Failed to generate comparator for scan value: %s", err)
-            elseif scan_comparator(scanner.after_tuple, scan_value) then
-                local after_tuple_key = utils.extract_key(scanner.after_tuple, index.parts)
-                scan_value = after_tuple_key
+            elseif scan_comparator(opts.after_tuple, scan_value) then
+                local after_tuple_key = utils.extract_key(opts.after_tuple, index.parts)
+                value = after_tuple_key
             end
         end
     end
 
     local tuple
-    local gen,param,state = index:pairs(scan_value, {iterator = scanner.iter})
+    local gen,param,state = index:pairs(value, {iterator = iter})
 
-    if scanner.after_tuple ~= nil then
+    if opts.after_tuple ~= nil then
         local err
-        tuple, err = scroll_to_after_tuple(gen, param, state, space, scanner)
+        tuple, err = scroll_to_after_tuple(gen, param, state, space, index, iter, opts.after_tuple)
         if err ~= nil then
-            return nil, ExecuteSelectError:new("Failed to scroll to the last tuple: %s", err)
+            return nil, ExecuteSelectError:new("Failed to scroll to the after tuple: %s", err)
```

Comment:
`the after tuple`... If you mean after_tuple you need underscore here `after tuple` -> `after_tuple`